eng/pipelines/templates/steps/retain-heap-dump-hprofs.yml

```diff
@@ -0,0 +1,17 @@
+steps:
+  - pwsh: |
+      New-Item $(Build.ArtifactStagingDirectory)/oom-hprofs -ItemType directory
+      foreach($hprofFile in (Get-ChildItem -Path . -Recurse -Filter *.hprof -File))
+      {
+        $fileFullName = $hprofFile.FullName
+        $fileName = $hprofFile.Name
+        Copy-Item $fileFullName $(Build.ArtifactStagingDirectory)/oom-hprofs/$fileName -ErrorAction SilentlyContinue
```

Comment:
How large are these files? Should we move them instead of copy?

Reply:
It'll be variable, and somewhat dependent on the max heap size, but I'd expect them to be in the high MBs, low GBs range. Moving them would likely be the better option in this case.